packages/sync-actions/test/price-sync.spec.js

```diff
@@ -925,4 +925,43 @@ describe('price actions', () => {
       ])
     })
   })
+
+  describe('removeStagedChanges', () => {
+    test('should build `removeStagedChanges` action when staged changes are removed', () => {
+      const before = {
+        id: '1234',
+        value: {
+          type: 'centPrecision',
+          currencyCode: 'USD',
+          centAmount: 5000,
+          fractionDigits: 2,
+        },
+        staged: {
+          type: 'centPrecision',
+          currencyCode: 'USD',
+          centAmount: 6000,
+          fractionDigits: 2,
+        },
+      }
+
+      const now = {
+        id: '1234',
+        value: {
+          type: 'centPrecision',
+          currencyCode: 'USD',
+          centAmount: 5000,
+          fractionDigits: 2,
+        },
+        staged: null,
```

Comment:
A couple of questions:

1. The object of the action `removeStagedChanges` does not include a `staged` property in the docs. Shouldn't we omit it?
2. Let's imagine we have the following now object:
```
const now = {
        id: '1234',
        value: {
          type: 'centPrecision',
          currencyCode: 'USD',
          centAmount: 5000,
          fractionDigits: 2,
        },
        staged: {
          type: 'centPrecision',
          currencyCode: 'EUR',
          centAmount: 5555,
          fractionDigits: 2,
        },
}
```
Which action is generated?
Is the

Reply:
@ragafus 
1. yes, removed staged property
2. Do you mean adding a `should not build removeStagedChanges action` test?

Reply:
> @ragafus
> 
> 1. yes, removed staged property
> 2. Do you mean adding a `should not build removeStagedChanges action` test?

1.  I'm not sure if having `staged: undefined` in the generated object is going to create an error on BE side.
2. Yes

Reply:
The staged property is not generated by the action anyway. I think the Jest equality matcher considered that both 
```
        {
          action: 'removeStagedChanges',
        }
```
```
         {
          action: 'removeStagedChanges',
          staged: undefined,
        }
```
are equal. That's why the assertion with the included `staged` property passes

2. Gonna add another test for it 👍

Reply:
> The staged property is not generated by the action anyway. I think the Jest equality matcher considered that both
> 
> ```
>         {
>           action: 'removeStagedChanges',
>         }
> ```
> 
> ```
>          {
>           action: 'removeStagedChanges',
>           staged: undefined,
>         }
> ```
> 
> are equal. That's why the assertion with the included `staged` property passes
> 
> 2. Gonna add another test for it 👍

1. You are right. Maybe we should switch to `toStrictEqual` matcher.

Reply:
Only having the `removeStagedChanges` action without the rest of the staged related actions is a bit confusing.

If the test data was like below, do we get a `removeStagedAction` and `changeValue` 🤔 It should be `applyStagedChanges` instead.

```
const now = {
    id: '1234',
    value: {
        type: 'centPrecision',
        currencyCode: 'USD',
        centAmount: 6000,
        fractionDigits: 2,
    },
};
```

And on @ragafus 's example, I would assume by looking at how other props are configured, we would get something like below 😕 
```
const action = {
    action: 'removeStagedChanges',
    staged: {
        type: 'centPrecision',
        currencyCode: 'EUR',
        centAmount: 5555,
        fractionDigits: 2,
    },
};
```

These staged related actions are designed implicitly stateful, seems to be not so compatible with the diff'er of sync-actions as is.

Reply:
@dogayuksel thx for your input, indeed the action builder doesn't get it right in the case you mentioned.
As far as I understand the situation and your comment, does that imply that the sync-actions we are using to determine an action aren't sufficient to handle this case?
Let's say I introduce `applyStagedChanges`
```
  { action: 'removeStagedChanges', key: 'staged' },
  { action: 'applyStagedChanges', key: 'staged' },
```
It doesn't seem to me that our action method can tell the difference... or am I missing something?